src/lm/arpa-file-parser.cc

```diff
@@ -147,6 +155,7 @@ void ArpaFileParser::Read(std::istream &is, bool binary) {
     int32 ngram_count = 0;
     while (++line_number_, getline(is, current_line_) && !is.eof()) {
       if (current_line_.empty()) continue;
+      current_line_ = rtrim(current_line_);
       if (current_line_[0] == '\\') break;
```

Comment:
Since you are looking at this... this code doesn't seem to be checking that the n-gram orders are as expected, it's just checking that the first character is backslash.  That's not very nice, and would lead to unexpected behavior if someone, say, skipped an ngram order.